src/tests/ftest/server/dynamic_start_stop.py

```diff
@@ -0,0 +1,122 @@
+
+  GOVERNMENT LICENSE RIGHTS-OPEN SOURCE SOFTWARE
+  The Government's rights to use, modify, reproduce, release, perform, display,
+  or disclose this software are subject to the terms of the Apache License as
+  provided in Contract No. B609815.
+  Any reproduction of computer software, computer software documentation, or
+  portions thereof marked with this legend must also reproduce the markings.
+"""
+import json
+from apricot import TestWithServers
+from dmg_utils import DmgCommand
+
+
+class DynamicStartStop(TestWithServers):
+    """Test Class Description:
+
+    SRS-10-0052 is about stopping the server and SRS-10-0053 is about starting,
+    so this test focuses on stopping and starting the new servers in an existing
+    system with a pool.
+
+    :avocado: recursive
+    """
+
+    def __init__(self, *args, **kwargs):
+        """Initialize a DynamicStartStop object."""
+        super(DynamicStartStop, self).__init__(*args, **kwargs)
+        self.dmg_cmd = None
+        self.stopped_ranks = set()
+
+    def verify_system_query(self):
+        """Verify state of the ranks.
+
+        Call dmg system query --json and verify the State of each rank. If the
+        rank is in self.stopped_ranks, verify that its status is Stopped.
+        Otherwise, Joined.
+        """
+        output = self.dmg_cmd.system_query().stdout
+        data = json.loads(output)
+        members = data["response"]["Members"]
+        for member in members:
+            if member["Rank"] in self.stopped_ranks:
+                self.assertEqual(
+                    member["State"], self.dmg_cmd.STOPPED,
+                    "State isn't Stopped! Actual: {}".format(member["State"]))
+                self.assertEqual(
+                    member["Info"], "system stop",
+                    "Info (Reason) isn't system stop! Actual: {}".format(
+                        member["Info"]))
+            else:
+                self.assertEqual(
+                    member["State"], self.dmg_cmd.JOINED,
+                    "State isn't Joined! Actual: {}".format(member["State"]))
+
+    def test_dynamic_server_addition(self):
+        """JIRA ID: DAOS-3598
+
+        Test Steps:
+        1. Start 2 servers and create a poll across these servers.
+        2. Start 3 additional servers and let them join the system. Verify with
+        dmg system query --verbose.
+        3. Stop one of the added servers - Single stop.
+        4. Stop two of the remaining added servers - Multiple stop.
+        5. Stop one of the original servers - Stopping with pool.
+
+        :avocado: tags=all,hw,large,server,full_regression,dynamic_start_stop
+        """
+        self.add_pool()
+
+        extra_servers = self.params.get("test_servers", "/run/extra_servers/*")
+        
+        # Start 3 extra servers.
+        self.start_additional_servers(additional_servers=extra_servers)
+
+        self.dmg_cmd = self.get_dmg_command()
+        self.dmg_cmd.json.update(True)
```

Comment:
Any reason we're not using a test yaml entry (`dmg: json: True`) to define this value?

Reply:
I didn't know we could do that. I tried it and it caused an error at self.add_pool(), so I added the support to parse json output in pool_create.